File: api-cpp/src/TimeUnit.cpp
/*
 *  Copyright (c) 2004 on behalf of the EU EGEE Project:
 *  The European Organization for Nuclear Research (CERN),
 *  Istituto Nazionale di Fisica Nucleare (INFN), Italy
 *  Datamat Spa, Italy
 *  Centre National de la Recherche Scientifique (CNRS), France
 *  CS Systeme d'Information (CSSI), France
 *  Royal Institute of Technology, Center for Parallel Computers (KTH-PDC), Sweden
 *  Universiteit van Amsterdam (UvA), Netherlands
 *  University of Helsinki (UH.HIP), Finland
 *  University of Bergen (UiB), Norway
 *  Council for the Central Laboratory of the Research Councils (CCLRC), United Kingdom
 */
#include "rgma/TimeUnit.h"

#include <sstream>
#include <string>

namespace glite {
namespace rgma {

const TimeUnit TimeUnit::SECONDS("SECONDS", 1);
const TimeUnit TimeUnit::MINUTES("MINUTES", 60);
const TimeUnit TimeUnit::HOURS("HOURS", 3600);
const TimeUnit TimeUnit::DAYS("DAYS", 86400);

TimeUnit::TimeUnit(std::string name, int ratio) :
    m_name(name), m_ratio(ratio) {
}

int TimeUnit::getSecs() const {
    return m_ratio;
}

std::ostream& operator<<(std::ostream& stream, const TimeUnit & tu) {
    return stream << tu.m_name;
}

}
}

File: api-java/src/org/glite/rgma/TimeUnit.java
/*
 * Copyright (c) Members of the EGEE Collaboration. 2004.
 * See http://eu-egee.org/partners/ for details on the copyright holders.
 * For license conditions see the license file or http://eu-egee.org/license.html
 */

package org.glite.rgma;

/**
 * Time units.
 */
public enum TimeUnit {

	/** Seconds. */
	SECONDS(1),

	/** Minutes. */
	MINUTES(60),

	/** Hours. */
	HOURS(3600),

	/** Days. */
	DAYS(86400);

	private int m_ratio;

	private TimeUnit(int ratio) {
		m_ratio = ratio;
	}

	/**
	 * Gets the number of seconds in this unit.
	 * 
	 * @return The number of seconds in one of this unit.
	 */
	 int getSecs() {
		return m_ratio;
	}
}
